test/Table_Tests/src/Common_Table_Operations/Join/Replace_Spec.enso

```diff
@@ -88,22 +94,65 @@ add_specs suite_builder setup =
 
         group_builder.specify "(edge-case) should allow lookup with itself" <|
             table = table_builder [['x', [2, 1, 4, 3]], ['y', [20, 10, 40, 30]]]
-            expected = table_builder [['x', [20, 10, 40, 30]], ['y', [20, 10, 40, 30]]]
-            result = table.replace table 'x'
+            expected = table_builder [['x', [10, 20, 30, 40]], ['y', [10, 20, 30, 40]]]
+            result = table.replace table 'x' . order_by "x"
             result . should_equal expected
 
         group_builder.specify "should not merge columns other than the one specified in the `column` param" <|
             table = table_builder [['x', [1, 2, 3, 4, 2]], ['y', ['a', 'b', 'c', 'd', 'e']], ['q', [4, 5, 6, 7, 8]]]
             lookup_table = table_builder [['x', [2, 1, 4, 3]], ['z', [20, 10, 40, 30]], ['q', [40, 50, 60, 70]]]
-            expected = table_builder [['x', [10, 20, 30, 40, 20]], ['y', ['a', 'b', 'c', 'd', 'e']], ['q', [4, 5, 6, 7, 8]]]
-            result = table.replace lookup_table 'x'
+            expected = table_builder [['x', [10, 20, 20, 30, 40]], ['y', ['a', 'b', 'e', 'c', 'd']], ['q', [4, 5, 8, 6, 7]]]
+            result = table.replace lookup_table 'x' . order_by ["x", "y"]
             result . should_equal expected
 
         group_builder.specify "should fail on null key values in lookup table" <|
             table = table_builder [['x', [1, 2, 3, 4, 2]], ['y', ['a', 'b', 'c', 'd', 'e']]]
             lookup_table = table_builder [['x', [2, 1, Nothing, 3]], ['z', [20, 10, 40, 30]]]
             table.replace lookup_table 'x' . should_fail_with Null_Values_In_Key_Columns
 
+        group_builder.specify "should fail on an empty lookup table with no columns" <|
+            table = table_builder [['x', [1, 2, 3, 4, 2]], ['y', ['a', 'b', 'c', 'd', 'e']]]
+            lookup_table = table_builder []
+            table.replace lookup_table 'x' . should_fail_with Illegal_Argument
+
+        group_builder.specify "should accept an empty (but well-formed) lookup table if allow_unmatched_rows=True, but expect a warning" <|
+            table = table_builder [['x', [1, 2, 3, 4, 2]], ['y', ['a', 'b', 'c', 'd', 'e']]]
+            lookup_table = table_builder_typed [['x', []], ['z', []]] Value_Type.Integer
+            t = table.replace lookup_table 'x' . order_by ['y']
+            t . should_equal table
+            Problems.expect_warning Empty_Error t
```

Comment:
```suggestion
            Problems.expect_warning (Empty_Error "lookup_table") t

```

If you go with my above change. But either way I think our tests should be more specific when testing errors.

Reply:
Done